R/read-write.R

```diff
@@ -0,0 +1,428 @@
+#' Read/write a local dm from/to files
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' Store a local `dm` in a file or directory and retrieve it later.
+#'
+#' @inheritParams dm_add_pk
+#' @param csv_directory For `dm_write_csv()`: The path to a non-existent or empty
+#' directory to write the `csv`-files defining the `dm` to.
+#'
+#' For `dm_read_csv()`: The path to the directory containing the `dm` as `csv-files`.
+#' @details `dm_write_csv()`: write a `dm` to a collection of `csv`-files to a non-existent or empty directory.
+#'
+#' @return `read`-family: A `dm` object.
+#'
+#' `write`-family: The path to the file/directory containing the stored `dm`, invisibly.
```

Comment:
Should return the input dm?

Reply:
I wouldn't mind that, and probably that makes more sense.
I somehow liked that it's pipeable (e.g. `dm_write_csv(dm_for_filter(), csv_directory = test_path) %>% dm_read_csv()`), but you're right, this will be rarely a use-case :)

I will change it to return the `dm`

Reply:
Yeah, this is also consistent with what `readr::write_csv()` does.

It's more useful to return the dm because writing to a file is a side effect, and the path can't really be computed on.